### PR TITLE
Add fs-gcsfs recipe

### PR DESCRIPTION
**Pull request by birnbaum**
Recipe for https://github.com/Othoz/gcsfs

### CODE REVIEW

File: recipes/fs-gcsfs/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  license_family: MIT
+  license_file: LICENSE
+  summary: A Python filesystem abstraction of Google Cloud Storage (GCS) implemented as a PyFilesystem2 extension
+  doc_url: https://fs-gcsfs.readthedocs.io/
+  dev_url: https://github.com/Othoz/gcsfs
+
+extra:
+  recipe-maintainers:
+    - birnbaum
+    - mathiaseitz
```
Comment (grst) [inline recipes/fs-gcsfs/meta.yaml]: @mathiaseitz, @trendelkampschroer, are you ok being added as maintainers?

Comment (trendelkampschroer) [reply recipes/fs-gcsfs/meta.yaml]: Yes, that's perfect. Thank you.

Comment (mathiaseitz) [reply recipes/fs-gcsfs/meta.yaml]: Yes. Thanks!

**Comment by grst** [comment]
@conda-forge/staged-recipes, LGTM

File: recipes/fs-gcsfs/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  noarch: python
+  script: {{ PYTHON }} -m pip install --no-deps --ignore-installed .
+
+requirements:
+  host:
+    - python >=3
+    - pip
+  run:
+    - python ==3.*, >=3.5
+    - google-cloud-storage ==1.*, >=1.10.0
```
[the remark this answers is not in the document]
Comment (birnbaum) [reply recipes/fs-gcsfs/meta.yaml]: just `>=1.10.0` would also allow e.g. `2.0.0` wouldn't it? And mayor version bumps will most likely introduce some breaking changes, so I thought it's a good idea to not allow them

[the remark this answers is not in the document]
Comment (ocefpaf) [reply recipes/fs-gcsfs/meta.yaml]: I'm not sure if conda accepts that syntax but @msarahan know better. In that case would use `>=1.10.0,<2`

[the remark this answers is not in the document]
Comment (birnbaum) [reply recipes/fs-gcsfs/meta.yaml]: we use this syntax in several projects and it seems to work fine, but of course I can change it to `<2` :)

**Comment by birnbaum** [comment]
Hey @conda-forge/staged-recipes, is this ready to merge? 🙂

**Pull request merged**